### ISSUE TITLE
Can't deploy karmada host on Azure Kubernetes Service with a LoadBalancer configuration

### ISSUE THREAD

**Issue by bcarlock-emerge** [karmada-io/karmada#2841]
I'm working on a proof of concept deployment in Azure Kubernetes Service of Karmada for cross region/multi-cloud HA. The Helm chart has some serious issues deploying functional services so I decided to try to install via script to determine the underlying issue. (The issue is at least in part due to the pre-deploy script failing to modify the CRDs, but I haven't gotten much past that)

**What happened**:  Services would deploy but once the karmada-api service deployed util::wait_service_external_ip consistently failed to detect the external IP. 
![image](https://user-images.githubusercontent.com/101663794/202944365-7010a5f2-c502-48f5-8f83-e4ffb634afa5.png)

**What you expected to happen**: Load balancer is detected and consumed as described in docs.

**How to reproduce it (as minimally and precisely as possible)**:  Install from local script to Azure AKS using the loadbalancer setting. 

**Anything else we need to know?**: I was able to fix the issue by bypassing util::wait_service_external_ip by manually setting SERVICE_EXTERNAL_IP to the service IP, commenting out the assignment of SERVICE_EXTERNAL_IP to ''  and then forcing util::wait_service_external_ip to return 0 (below)
`export SERVICE_EXTERNAL_IP=<EXTERNALIP>`
```
# util::wait_service_external_ip give a service external ip when it is ready, if not, wait until timeout
# Parameters:
#  - $1: context name in k8s
#  - $2: service name in k8s
#  - $3: namespace
#SERVICE_EXTERNAL_IP=''
function util::wait_service_external_ip() {
  local context_name=$1
  local service_name=$2
  local namespace=$3
  local external_ip
  local tmp
#  for tmp in {1..30}; do
#    set +e
#    external_host=$(kubectl --context="$context_name" get service "${service_name}" -n "${namespace}" --template="{{range .status.loadBalancer.ingress}}{{.hostname}} {{end}}" | xargs)
#    external_ip=$(kubectl --context="$context_name" get service "${service_name}" -n "${namespace}" --template="{{range .status.loadBalancer.ingress}}{{.ip}} {{end}}" | xargs)
#    set -e
#    if [[ ! -z "$external_host" ]]; then # Compatibility with hostname, such as AWS
#      external_ip=$external_host
#    fi
#    if [[ -z "$external_ip" ]]; then
#      echo "wait the external ip of ${service_name} ready..."
#      sleep 6
#      continue
#    else
#      SERVICE_EXTERNAL_IP="${external_ip}"
#      return 0
#    fi
#  done
  return 0
}
```

**Environment**:
- Karmada version: v1.3.1
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`): karmadactl version: version.Info{GitVersion:"v1.3.1", GitCommit:"1fe31b182a6d251e09528530ec51de8adb42d202", GitTreeState:"clean", BuildDate:"2022-10-10T09:34:39Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
- Others:

**Comment by RainbowMango** [comment]
Thanks @bcarlock-emerge.
cc @Poor12 Could you please confirm it?

**Comment by Poor12** [comment]
From the logs, "error: context 'karmada-apiserver does not exist", we can know that there is failure when `kubectl --context karmada-apiserver get svc`.

It's a little strange. I wonder if you have modified the scripts or there is a confusing kubeconfig.

**Comment by bcarlock-emerge** [comment]
No, I hadn't modified any scripts until the change I posted. This was run from a fresh clone of the master branch.
My kubeconfig isn't anything special, I have 9 clusters in it 3 of which are for this proof of concept. 
Kubectl --context works as expected and works to set up the earlier parts of the deployment. Is there some way I can provide additional diagnostic information for you?

**Comment by bcarlock-emerge** [comment]
At what point should the the karmada-apiserver context be added to my config?

**Comment by Tanyril** [comment]
Here's the full dump of my session last night with a couple of values redacted. 
[KarmadaInstallShell.txt](https://github.com/karmada-io/karmada/files/10057457/KarmadaInstallShell.txt)

**Comment by bcarlock-emerge** [comment]
Here's my full session from yesterday with a few values redacted. There are few points I marked with <----------------------------- for ease of navigation.
[KarmadaInstallShell.txt](https://github.com/karmada-io/karmada/files/10057474/KarmadaInstallShell.txt)

**Comment by Poor12** [comment]
Hi @bcarlock-emerge, I think I have found the problem.
You can fix it with #2844 by the following command:
```
export LOAD_BALANCER=true
./hack/deploy-karmada.sh ~/.kube/members.config member1 remote
```

**Comment by bcarlock-emerge** [comment]
I've moved on to helm/gitops deployment because I was able to work around the issue.
